ach/src/main/java/com/adyen/checkout/ach/internal/ui/StoredACHDirectDebitDelegate.kt

```diff
@@ -0,0 +1,190 @@
+import com.adyen.checkout.components.core.StoredPaymentMethod
+import com.adyen.checkout.components.core.internal.PaymentComponentEvent
+import com.adyen.checkout.components.core.internal.PaymentObserverRepository
+import com.adyen.checkout.components.core.internal.data.api.AnalyticsRepository
+import com.adyen.checkout.components.core.internal.ui.model.FieldState
+import com.adyen.checkout.components.core.internal.ui.model.Validation
+import com.adyen.checkout.components.core.internal.util.PaymentMethodTypes
+import com.adyen.checkout.components.core.internal.util.bufferedChannel
+import com.adyen.checkout.components.core.paymentmethod.ACHDirectDebitPaymentMethod
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.internal.util.LogUtil
+import com.adyen.checkout.core.internal.util.Logger
+import com.adyen.checkout.ui.core.internal.ui.AddressFormUIState
+import com.adyen.checkout.ui.core.internal.ui.ButtonComponentViewType
+import com.adyen.checkout.ui.core.internal.ui.ComponentViewType
+import com.adyen.checkout.ui.core.internal.ui.PaymentComponentUIEvent
+import com.adyen.checkout.ui.core.internal.ui.PaymentComponentUIState
+import com.adyen.checkout.ui.core.internal.ui.SubmitHandler
+import com.adyen.checkout.ui.core.internal.ui.model.AddressInputModel
+import com.adyen.checkout.ui.core.internal.ui.model.AddressOutputData
+import com.adyen.checkout.ui.core.internal.util.AddressValidationUtils
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+@Suppress("TooManyFunctions")
+internal class StoredACHDirectDebitDelegate(
+    private val observerRepository: PaymentObserverRepository,
+    private val storedPaymentMethod: StoredPaymentMethod,
+    private val analyticsRepository: AnalyticsRepository,
+    override val componentParams: ACHDirectDebitComponentParams,
+    private val order: OrderRequest?,
+    private val submitHandler: SubmitHandler<ACHDirectDebitComponentState>,
+) : ACHDirectDebitDelegate {
+
+    private val inputData: ACHDirectDebitInputData = ACHDirectDebitInputData()
+
+    private val _outputDataFlow = MutableStateFlow(createOutputData())
+    override val outputDataFlow: Flow<ACHDirectDebitOutputData> = _outputDataFlow
+
+    override val outputData: ACHDirectDebitOutputData
+        get() = _outputDataFlow.value
+
+    private val _componentStateFlow = MutableStateFlow(createComponentState())
+    override val componentStateFlow: Flow<ACHDirectDebitComponentState> = _componentStateFlow
+
+    private val _viewFlow: MutableStateFlow<ComponentViewType?> = MutableStateFlow(ACHDirectDebitComponentViewType)
```

Comment:
Is the `ACHDirectDebitComponentViewType` needed here or can we just set `MutableStateFlow(null)`?

Reply:
If merchants use the standalone stored component it will show an empty view now while it actually shouldn't, right?